src/test/java/seedu/address/logic/commands/DeleteStudentCommandTest.java

```diff
@@ -5,58 +5,152 @@
+                + "sn/STUDENT NUMBER to delete the student.", () ->
+                command.execute(modelStub));
+    }
+
+    // Ensure deletion by only provided name works when student with correct details exists
     @Test
-    public void execute_studentDelete_success() throws Exception {
+    public void execute_singleStudentDeleteByNameOnly_success() throws Exception {
         Student validStudent = new StudentBuilder().withName("John Ng").build();
         DeleteStudentCommandTest.ModelStubWithStudent modelStub =
                 new DeleteStudentCommandTest.ModelStubWithStudent(validStudent);
-        modelStub.addStudent(validStudent);
 
         DeleteStudentCommand command = new DeleteStudentCommand(new Name("John Ng"));
 
         CommandResult result = command.execute(modelStub);
 
-        assertEquals(String.format(DeleteStudentCommand.MESSAGE_DELETE_STUDENT_SUCCESS + validStudent.getName()),
+        assertEquals(String.format(DeleteStudentCommand.MESSAGE_DELETE_STUDENT_SUCCESS + validStudent.getName()
+                + " " + validStudent.getStudentNumber()),
                 result.getFeedbackToUser());
     }
 
+    // Ensure deletion by provided name and provided student number works when student with correct details exists
+    @Test
+    public void execute_singleStudentDeleteByNameAndStudentNumber_success() throws Exception {
+        Student validStudent = new StudentBuilder().withName("John Ng").withStudentNumber("A1234567X").build();
+        DeleteStudentCommandTest.ModelStubWithStudent modelStub =
+                new DeleteStudentCommandTest.ModelStubWithStudent(validStudent);
+
+        DeleteStudentCommand command = new DeleteStudentCommand(new Name("John Ng"), new StudentNumber("A1234567X"));
+
+        CommandResult result = command.execute(modelStub);
+
+        assertEquals(String.format(DeleteStudentCommand.MESSAGE_DELETE_STUDENT_SUCCESS + validStudent.getName()
+                        + " " + validStudent.getStudentNumber()),
+                result.getFeedbackToUser());
+    }
+
+    // Ensure deletion by provided name and provided student number works with duplicate student names
+    @Test
+    public void execute_duplicateStudentDeleteByNameAndStudentNumber_success() throws Exception {
+        Student validStudent1 = new StudentBuilder().withName("John Ng").withStudentNumber("A1234567X").build();
+        Student validStudent2 = new StudentBuilder().withName("John Ng").withStudentNumber("A0123456Y").build();
+        Student validStudent3 = new StudentBuilder().withName("John Ng").withStudentNumber("A9876543Z").build();
+        Student validStudent4 = new StudentBuilder().withName("John Ng").withStudentNumber("A1111111B").build();
+        DeleteStudentCommandTest.ModelStubWithStudent modelStub =
+                new DeleteStudentCommandTest.ModelStubWithStudent(validStudent1, validStudent2,
+                        validStudent3, validStudent4);
+
```

Comment:
Perhaps you should consider covering the part of the code where they just put the name which will raise the CommandException that tells the user that there are multiple students with the same name.